kura/org.eclipse.kura.misc.cloudcat/src/main/java/org/eclipse/kura/internal/misc/cloudcat/CloudCat.java

```diff
@@ -0,0 +1,234 @@
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ *******************************************************************************/
+package org.eclipse.kura.internal.misc.cloudcat;
+
+import static org.eclipse.kura.configuration.ConfigurationService.KURA_SERVICE_PID;
+
+import java.util.Map;
+
+import org.eclipse.kura.KuraException;
+import org.eclipse.kura.cloud.CloudClient;
+import org.eclipse.kura.cloud.CloudService;
+import org.eclipse.kura.configuration.ConfigurableComponent;
+import org.osgi.framework.Filter;
+import org.osgi.framework.InvalidSyntaxException;
+import org.osgi.framework.ServiceReference;
+import org.osgi.service.component.ComponentContext;
+import org.osgi.util.tracker.ServiceTracker;
+import org.osgi.util.tracker.ServiceTrackerCustomizer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CloudCat implements ConfigurableComponent {
+
+    private static final Logger logger = LoggerFactory.getLogger(CloudCat.class);
+
+    private CloudCatOptions options;
+    private CloudService firstCloudService;
+    private CloudService secondCloudService;
+    private CloudClient firstCloudClient;
+    private CloudClient secondCloudClient;
+    private CloudClientRelay firstCloudClientRelay;
+    private CloudClientRelay secondCloudClientRelay;
+
+    private ServiceTracker<CloudService, CloudService> firstCloudServiceTracker;
+    private ServiceTracker<CloudService, CloudService> secondCloudServiceTracker;
+
+    private ComponentContext componentContext;
+
+    public void bindFirstCloudService(CloudService firstCloudService) {
+        this.firstCloudService = firstCloudService;
+
+        try {
+            this.firstCloudClient = this.firstCloudService.newCloudClient(this.options.getFirstCloudClientAppId());
+            initClients();
+        } catch (KuraException e) {
+            logger.error("CloudClient: {} instantiation failed", this.options.getFirstCloudClientAppId(), e);
+        }
+    }
+
+    public void bindSecondCloudService(CloudService secondCloudService) {
+        this.secondCloudService = secondCloudService;
+
+        try {
+            this.secondCloudClient = this.secondCloudService.newCloudClient(this.options.getSecondCloudClientAppId());
+            initClients();
+        } catch (KuraException e) {
+            logger.error("CloudClient: {} instantiation failed", this.options.getSecondCloudClientAppId(), e);
+        }
+    }
+
+    public void unbindFirstCloudService(CloudService firstCloudService) {
+        cleanupClients();
+        this.firstCloudService = null;
+    }
+
+    public void unbindSecondCloudService(CloudService firstCloudService) {
+        cleanupClients();
+        this.secondCloudService = null;
+    }
+
+    protected void activate(ComponentContext ctx, Map<String, Object> properties) {
+        logger.info("Activating");
```

Comment:
hello @amitjoy what do you suggest? The name of the class is already resent in what the logger will print. I can add the CloudCat instance ``logger.info("Activating {}", this)``...

Reply:
You are right that the name would be displayed in the log as a constituent of the logger instance. I believe it would be more readable if we can provide the component's user-friendly name as well.

Activating CloudCat, Deactivating CloudCat and so on.